src/app/shared/directives/border-decorator/border-decorator.directive.ts

```diff
@@ -0,0 +1,23 @@
+import { Directive, ElementRef, Input, OnInit } from "@angular/core";
+// import { CourseComponent } from "../course.component";
+import { generateBorderColor } from "../../../core/helpers/index";
+
+@Directive({
+    selector: "[appBorderDecorator]"
+})
+export class BorderDecoratorDirective implements OnInit {
+    private readonly element: ElementRef;
+
+    /**
+     * input field for directive that receives Date
+     */
+    @Input() public dateToColor: Date;
+
+    constructor(element: ElementRef) {
+        this.element = element;
+    }
+
+    public ngOnInit(): void {
+        this.element.nativeElement.style.border = generateBorderColor(this.dateToColor);
```

Comment:
1.) You should avoid using base DOM api. It should be done with Renderer2
2.) Renderer2 should be used ONLY in exceptional cases.
3.) In current case you have 3 color options, so it can be done:
- ngClass
- HostBinding

The preferable one is HostBinding

Reply:
done